File: csharp/Main/EmailSender.cs
using System;

namespace Soat.CleanCoders.DipKata.Main
{
    public class EmailSender
    {
        public void send(string emailAddress, string message)
        {
            Console.Write($"To:{emailAddress}, Subject: Happy birthday!, Message: {message}");
        }
    }
}

File: java/src/test/java/fr/soat/cleancoders/details/EmailSender.java
package fr.soat.cleancoders.details;


import fr.soat.cleancoders.highlevel.Friend;
import fr.soat.cleancoders.highlevel.MessageSender;

public class EmailSender implements MessageSender {

    @Override
    public void send(Friend aFriend) {
        System.out.print("To:" + aFriend.getEmailAdress() + ", Subject: Happy birthday!" + ", Message: "
                + messageForA(aFriend.getName()));
    }

    private String messageForA(String name) {
        return String.format("Happy birthday, dear %s!", name);
    }
}
